e2e/testcases/reconciler_manager_test.go

```diff
@@ -150,7 +150,11 @@ func TestManagingReconciler(t *testing.T) {
 	nt.T.Log("Switch the auth type from none to gcpserviceaccount")
 	nt.MustMergePatch(rs, `{"spec":{"git":{"auth":"gcpserviceaccount","secretRef":{"name":""},"gcpServiceAccountEmail":"test-gcp-sa-email@test-project.iam.gserviceaccount.com"}}}`)
 	nt.T.Log("Verify the gcenode-askpass-sidecar container should be added")
-	generation++
+	if nt.IsGKEAutopilot {
+		generation += 2 // generation bumped by 2 because the sidecar container will first be added and then the resource requirements will be adjusted by autopilot
```

Comment:
Do we know that autopilot will always adjust the resource requirements exactly once?

Reply:
Yes, the reconciler-manager will [keep the resource requirements adjusted by Autopilot](https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/reconcilermanager/controllers/reconciler_base.go#L313). If it already follows the Autopilot rules, no more adjustment will be performed by Autopilot.